bundles/org.eclipse.swt/Eclipse SWT Browser/win32/org/eclipse/swt/browser/Edge.java

```diff
@@ -373,14 +379,30 @@ WebViewEnvironment createEnvironment() {
 
 @Override
 public void create(Composite parent, int style) {
-	checkDeadlock();
 	containingEnvironment = createEnvironment();
 
 	long[] ppv = new long[1];
 	int hr = containingEnvironment.environment().QueryInterface(COM.IID_ICoreWebView2Environment2, ppv);
 	if (hr == COM.S_OK) environment2 = new ICoreWebView2Environment2(ppv[0]);
 
+	// If leads to deadlock then execute asynchronously and move on.
+	// The webview calls are queued to be executed when it is done executing the current task.
+	if (leadsToDeadlock() && !asyncExecution) {
```

Comment:
I do no understand the reasons for the `asyncExecution` field. Its evaluation would only make a difference in case asynchronous creation is started for an instance of `Edge` and within _the same_ instance another `create` call is performed. Can that happen? I only see a single caller of `create()`, which is the instantiation of `Browser`. Am I missing something?

Reply:
Nope, you are right, I refactored this block adn moved some stuff to setupBrowser and forgot to remove this check. Sorry, I'll remove it.